src/cript/api/paginator.py

```diff
@@ -123,6 +126,7 @@ def current_page_number(self) -> int:
         return self._current_page_number
 
     @current_page_number.setter
+    @beartype
```

Comment:
we have to at some point write a wiki about beartype, why we used it, what it does, why we put it specifically always under the `property` decorator, so that anyone maintaining/helping with it will not get themselves into trouble later and create issues

Reply:
yeah, that makes sense. if we stick with beartype